src/agentscope/models/yi_model.py

```diff
@@ -0,0 +1,184 @@
+
+    def _register_default_metrics(self) -> None:
+        # Set monitor accordingly
+        self.monitor.register(
+            self._metric("call_counter"),
+            metric_unit="times",
+        )
+        self.monitor.register(
+            self._metric("prompt_tokens"),
+            metric_unit="token",
+        )
+        self.monitor.register(
+            self._metric("completion_tokens"),
+            metric_unit="token",
+        )
+        self.monitor.register(
+            self._metric("total_tokens"),
+            metric_unit="token",
+        )
+
+    def format(
+        self,
+        *args: Union[Msg, Sequence[Msg]],
+    ) -> Union[List[dict], str]:
+        raise NotImplementedError(
+            f"Model Wrapper [{type(self).__name__}] doesn't "
+            f"implement the format method. Please implement it "
+            f"in the subclass.",
+        )
+
+
+class YiChatWrapper(YiWrapperBase):
+    """The model wrapper for Yi's chat API."""
+
+    model_type: str = "yi_chat"
+
+    def __call__(
+        self,
+        messages: list,
+        **kwargs: Any,
+    ) -> ModelResponse:
+        """Processes a list of messages and makes a request to the Yi API."""
+        # Prepare keyword arguments
+        kwargs = {**self.generate_args, **kwargs}
+
+        # Checking messages
+        if not isinstance(messages, list):
+            raise ValueError(
+                f"Yi `messages` field expected type `list`, "
+                f"got `{type(messages)}` instead.",
+            )
+        if not all("role" in msg and "content" in msg for msg in messages):
+            raise ValueError(
+                "Each message in the 'messages' list must contain a 'role' "
+                "and 'content' key for Yi API.",
+            )
+
+        # Forward to generate response
+        response = self.client.chat.completions.create(
+            model=self.model_name,
+            messages=messages,
+            **kwargs,
+        )
+
+        # Record the api invocation if needed
+        self._save_model_invocation(
+            arguments={
+                "model": self.model_name,
+                "messages": messages,
+                **kwargs,
+            },
+            response=response.model_dump(),
+        )
+
+        # Update monitor accordingly
+        self.update_monitor(call_counter=1, **response.usage.model_dump())
+
+        # Return response
+        return ModelResponse(
+            text=response.choices[0].message.content,
+            raw=response.model_dump(),
+        )
+
+    def format(
+        self,
+        *args: Union[Msg, Sequence[Msg]],
+    ) -> List[dict]:
+        """Format the input messages for the Yi Chat API."""
+        messages = []
+        for arg in args:
+            if arg is None:
+                continue
+            if isinstance(arg, Msg):
+                messages.append(
+                    {
+                        "role": arg.role,
```

Comment:
The `name` field is not used in this format function, so that the llm doesn't know who is talking in a multi-agent conversation. 

Please refer to the format function of DashScopeChatWrapper, which combine the chat history into one system message.